esp8266/modnetwork.c

```diff
@@ -109,6 +110,13 @@ STATIC mp_obj_t esp_connect(mp_uint_t n_args, const mp_obj_t *args) {
             p = "";
         }
         memcpy(config.password, p, len);
+        if (n_args > 3) {
+            bssid = mp_obj_str_get_data(args[3], &len);
+            config.bssid_set = 1;
+        } else {
+            bssid = "";
+        }
+        memcpy(config.bssid, bssid, len);
```

Comment:
config is already zero'd on creation so you don't need to memcpy if the bssid arg is not given.  You can simplify this code to just a simple if clause with no else (and the bssid variable can be declared in the if).  Also, you need to truncate the string if it's longer than the room in the config.bssid array.